tokio-macros/src/entry.rs

```diff
@@ -363,7 +363,13 @@ fn parse_knobs(mut input: syn::ItemFn, is_test: bool, config: FinalConfig) -> To
             #crate_ident::runtime::Builder::new_current_thread()
         },
         RuntimeFlavor::Threaded => quote_spanned! {last_stmt_start_span=>
-            #crate_ident::runtime::Builder::new_multi_thread()
+            #[cfg(target_os = "wasi")]
+            let a = #crate_ident::runtime::Builder::new_current_thread();
+
+            #[cfg(not(target_os = "wasi"))]
+            let a = #crate_ident::runtime::Builder::new_multi_thread();
+
+            a
```

Comment:
It is better to implement the macro in a way such that it emits an error if you try to use the multi-threaded runtime, instead of just silently using the current-thread runtime instead. You can do this by modifying the `rt_multi_thread_available` and `default_flavor` fields here:

https://github.com/tokio-rs/tokio/blob/8aa55f1eacc2fd8930f4f0855441845feddc6482/tokio-macros/src/entry.rs#L56-L60

Then introduce a wasm-specific error here:

https://github.com/tokio-rs/tokio/blob/8aa55f1eacc2fd8930f4f0855441845feddc6482/tokio-macros/src/entry.rs#L144-L151